proxy-libs/heliossqlx/heliossqlx_test.go

```diff
@@ -0,0 +1,72 @@
+package heliossqlx
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uptrace/opentelemetry-go-extra/otelsql"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	_ "modernc.org/sqlite"
+)
+
+func getSpanRecorder() *tracetest.SpanRecorder {
+	spanRecorder := tracetest.NewSpanRecorder()
+	otel.SetTracerProvider(sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(spanRecorder)))
+	propagator := propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{})
+	otel.SetTextMapPropagator(propagator)
+	return spanRecorder
+}
+
+func assertSpan(t *testing.T, spanRecorder *tracetest.SpanRecorder) []attribute.KeyValue {
+	spanRecorder.ForceFlush(context.Background())
+	spans := spanRecorder.Ended()
+	assert.Equal(t, 2, len(spans))
+	spanDbConnect := spans[0]
+	assert.Equal(t, "db.Connect", spanDbConnect.Name())
+	assert.Equal(t, trace.SpanKindClient, spanDbConnect.SpanKind())
+
+	spanDbQuery := spans[1]
+	assert.Equal(t, "db.Query", spanDbQuery.Name())
+	assert.Equal(t, trace.SpanKindClient, spanDbQuery.SpanKind())
+	return spanDbQuery.Attributes()
+}
+
+func assertAttributes(t *testing.T, attributes []attribute.KeyValue) {
+	for _, attribute := range attributes {
+		key := attribute.Key
+		value := attribute.Value.AsString()
+
+		switch key {
+		case semconv.DBNameKey:
+			assert.Equal(t, "test", value)
+		case semconv.DBSystemKey:
```

Comment:
Do we get the db.system without the additional attribute that's sent to `Open`?

Reply:
yes :)

Reply:
Ok so let's remove the additional options

Reply:
but if the customer will want to add attributes?

Reply:
sorry I didn't read your question correctly and the answer is actually no, if we are not setting the db.system it doesnt infer this automatically

Reply:
Why not take it from the first argument?

Reply:
We can but it won't remove the need to add the options (for db.name for example) wdyt?

Reply:
Don't remove the options, but make sure that we collect the `db.system` without it